test/models/user_test.rb

```diff
@@ -4,12 +4,12 @@
 
 class UserTest < ActiveSupport::TestCase
   test '名前がないユーザはメールアドレスを返す' do
-    yocchan = build(:yocchan)
-    assert_equal yocchan.email, yocchan.name_or_email
+    yocchan = build(:user, email: 'yocchan@example.com', name: '')
+    assert_equal 'yocchan@example.com', yocchan.name_or_email
   end
 
   test '名前があるユーザは名前を返す' do
-    takeru = build(:takeru)
+    takeru = build(:user, name: 'たける')
```

Comment:
emailはFactoryBot側で設定されますが、ここで直接書いた方が定義を見に行かなくて済むので可読性が上がります。

```suggestion
    takeru = build(:user, email: 'takeru@example.com', name: 'たける')
```